lib/engine/tile.rb

```diff
@@ -243,6 +244,10 @@ def terrain
       @upgrades.flat_map(&:terrains).uniq
     end
 
+    def opposite!(tile)
```

Comment:
uhhh why? just make opposite assecor and do tile.opposite = x

Reply:
Sure.